### DISCUSSION TITLE
Is there a way of getting the rendered markdown from a subtree of SyntaxTreeNodes?

### DISCUSSION THREAD

**Discussion by anapaulagomes** [executablebooks/markdown-it-py#297] [Q&A]
Hi folks! First of all: really nice library. Thanks for putting it together!

I'd like to get part of a subtree of SyntaxTreeNodes and render it back to the markdown format (HTML would do the job too, I think). Is there a way of doing it? I've been exploring with tokens and such but I couldn't find a way of doing it yet.

Example:

I have this tree of SyntaxTreeNodes from the markdown below:

```
## Checklist

1. Pull the code from the `main` branch
from the main project repo
2. Configure the environment
    1. Configure the environment variables in the `.env` file
    2. Install the dependencies with:
   bash
   pip install -r requirements.txt

3. Run the migrations: `python manage.py migrate`
4. Done!
```

When I get to point 2, I'd like to have its inner enumerated list as markdown, like:

```
    1. Configure the environment variables in the `.env` file
    2. Install the dependencies with:
   bash
   pip install -r requirements.txt

```

How could I do it?

Any help is appreciated! Thanks! 🌷

**Comment by chrisjsewell** [comment]
Heya:

```python
from markdown_it import MarkdownIt
from markdown_it.tree import SyntaxTreeNode

text = """
## Checklist

1. Pull the code from the `main` branch
from the main project repo
2. Configure the environment
    1. Configure the environment variables in the `.env` file
    2. Install the dependencies with:
   bash
   pip install -r requirements.txt

3. Run the migrations: `python manage.py migrate`
4. Done!
"""

env = {}
md = MarkdownIt()
node = SyntaxTreeNode(md.parse(text, env))
sublist = node.children[1].children[1].children[1]
html = md.renderer.render(sublist.to_tokens(), md.options, env)
print(html)
```

For rendering back to Markdown, there is: https://github.com/executablebooks/mdformat/blob/0cbd2054dedf98ec8366001c8a16eacfa85cebc1/src/mdformat/renderer/__init__.py#L27

```python
from mdformat.renderer import MDRenderer

renderer = MDRenderer()
markdown = renderer.render(sublist.to_tokens(), md.options, env)
print(markdown)
```

**Reply by anapaulagomes** [reply]
That's really cool, @chrisjsewell! Thank you for your reply. I have managed to get it as HTML, altough having it as markdown is way better. But it seems that, based on your example and `render()`'s docstring, it wouldn't be possible to get any subtree, only from the grandchildren (like in your example). I did a bit of hacking to do it (posting here to get your feedback and maybe others in the future):

```
from markdown_it import MarkdownIt
from mdformat.renderer import MDRenderer

md = MarkdownIt("commonmark").enable("strikethrough")
renderer = MDRenderer()

def extract_markdown_from_subtree(node):
    # make it a tree first
    node.token, node.nester_tokens = None, []
    return renderer.render(node.to_tokens(), md.options, {})
```

And this is how I was getting the HTML

```
from markdown_it.renderer import RendererHTML

rh = RendererHTML()
rh.render(node.to_tokens(), md.options,{})
```

Thanks!
